doc/adrs/195_add_impact_analysis_for_contextual_menu_actions_of_trees.adoc

```diff
@@ -0,0 +1,98 @@
+= ADR-195 - Add impact analysis for contextual menu actions of trees
+
+== Context
+
+A tree contextual menu action can trigger an impact analysis to let the user decide whether they want to execute the tool.
+
+== Decision
+
+We will add an API to run impact analysis on trees and factorize common code between diagram and tree impact analysis.
+
+
+=== Frontend
+
+We will move the following types from `sirius-components-diagrams` to `sirius-components-core` to make the impact analysis dialog reusable in non-diagram contexts:
+
+* ImpactAnalysisDialogContext
+* ImpactAnalysisDialogContext.types
+* useImpactAnalysisDialog
+* useImpactAnalysisDialog.types
+
+The variables `diagramElementId` in these types will be renamed to `elementId` to decouple them from diagrams.
+
+We will use the `ImpactAnalysisDialogContext` in `TreeView` to make it visible to tree contextual menu actions:
```

Comment:
Or we will put this context even higher in the workbench or edit project view directly

Reply:
fixed